File: 2021/1.py
'''
Partial sums
'''
def main():
    with open("inputs/1.txt", 'r') as file:
        filedump = file.read()

    numbers = [int(num) for num in filedump.split('\n')]

    incs = 0
    for ind in range(1, len(numbers)):
        if numbers[ind] > numbers[ind-1]:
            incs += 1
    print("First part: ", incs)

    incs = 0
    for ind in range(3, len(numbers)):
        if numbers[ind] > numbers[ind-3]:
            incs += 1
    print("Second part: ", incs)


if __name__ == "__main__":
    main()

File: 2021/1.jl
using DelimitedFiles

numbers = readdlm("inputs/1.txt", '\n', Int)

incs = 0
for ind = 2:length(numbers)
    if numbers[ind] > numbers[ind-1]
        global incs += 1
    end
end
print("First part: ", incs)

incs = 0
for ind = 4:length(numbers)
    if numbers[ind] > numbers[ind-3]
        global incs += 1
    end
end
print("\nSecond part: ", incs)
